tests/mason-lspconfig/api/api_spec.lua

```diff
@@ -0,0 +1,55 @@
+local mason_lspconfig = require "mason-lspconfig"
+local _ = require "mason-core.functional"
+
+describe("mason-lspconfig API", function()
+    it("should return all available servers", function()
+        local server_mappings = require "mason-lspconfig.mappings.server"
+        local available_servers = mason_lspconfig.get_available_servers()
+        assert.equal(#vim.tbl_keys(server_mappings.package_to_lspconfig), #available_servers)
+    end)
+
+    it("should return all available servers for given filetype", function()
+        assert.same(
+            { "terraformls", "tflint" },
+            _.sort_by(
+                _.identity,
+                mason_lspconfig.get_available_servers {
+                    filetype = "terraform",
+                }
+            )
+        )
+    end)
+
+    it("should return all available servers for given filetypes", function()
+        assert.same(
+            { "lemminx", "taplo" },
+            _.sort_by(
+                _.identity,
+                mason_lspconfig.get_available_servers {
+                    filetype = { "xml", "xsd", "xsl", "toml" },
+                }
+            )
+        )
+    end)
+
+    it("should return no servers if filetype predicate has no matches", function()
+        assert.same(
+            {},
+            mason_lspconfig.get_available_servers {
+                filetype = { "thisfiletypesimplydoesntexist" },
+            }
+        )
+    end)
+
+    it("should handle servers without a matching registry", function()
```

Comment:
I stumbled upon this test-case while trying a few things, but I'm not sure what the behavior should be.

do you think it's worth it? feels like it would be useful for the CI at least, or is this out-of-scope for mason-lspconfig itself?